sdk/src/logs/simple_log_processor.cc

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/sdk/logs/simple_log_processor.h"
+
+#include <chrono>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+/**
+ * Initialize a simple log processor.
+ * @param exporter the exporter where log records are pushed by this processor
+ */
+SimpleLogProcessor::SimpleLogProcessor(std::unique_ptr<LogExporter> &&exporter)
+    : exporter_(std::move(exporter))
+{}
+
+// Batches the log record it receives in a batch of 1 and immediately sends it to the configured
+// exporter
+void SimpleLogProcessor::OnReceive(
+    std::unique_ptr<opentelemetry::logs::LogRecord> &&record) noexcept
+{
+  std::vector<std::unique_ptr<opentelemetry::logs::LogRecord>> batch;
+  batch.emplace_back(std::move(record));
+
+  // Get lock to ensure Export() is never called concurrently
+  std::lock_guard<opentelemetry::common::SpinLockMutex> locked(lock_);
+
+  // Check the ExportCode returned from the export
+  if (exporter_->Export(batch) == ExportResult::kFailure)
+  {
+    /* TODO: alert user of the failed export result */
+  }
+}
+// The simple processor does not have any log records to flush so this method is not used
+ProcessorResult SimpleLogProcessor::ForceFlush(std::chrono::microseconds timeout) noexcept
+{
+  // This function and thus return valie is unused
+  return ProcessorResult::kSuccess;
+}
+
+// TODO: abort functionality is currently not implemented if the processor times out
+ProcessorResult SimpleLogProcessor::Shutdown(std::chrono::microseconds timeout) noexcept
```

Comment:
Options to abort: 
1. Another thread - tell that thread to die when its able to 
2. Have an interrupt (not a good idea) 
3. Pull for a timer, mutex

Reply:
Basically options that make sense are a) add complexity to use thread to make log processor able to enforce timeout b) or trust exporters honor the timeout.